text/profiles/0239-profiles-data-model.md

```diff
@@ -0,0 +1,1611 @@
+  // If the original payload is in pprof format, it SHOULD not be included in this field.
+  // The field is optional, however if it is present `profile` MUST be present and contain the same profiling information.
+  bytes original_payload = 7;
+
+  // This is a reference to a pprof profile. Required, even when original_payload is present.
+  opentelemetry.proto.profiles.v1.alternatives.pprofextended.Profile profile = 8;
+}
+```
+
+<!-- proto1 -->
+
+Modified pprof:
+
+<!-- proto2 -->
+
+```proto
+// Profile is a common stacktrace profile format.
+//
+// Measurements represented with this format should follow the
+// following conventions:
+//
+// - Consumers should treat unset optional fields as if they had been
+//   set with their default value.
+//
+// - When possible, measurements should be stored in "unsampled" form
+//   that is most useful to humans.  There should be enough
+//   information present to determine the original sampled values.
+//
+// - On-disk, the serialized proto must be gzip-compressed.
+//
+// - The profile is represented as a set of samples, where each sample
+//   references a sequence of locations, and where each location belongs
+//   to a mapping.
+// - There is a N->1 relationship from sample.location_id entries to
+//   locations. For every sample.location_id entry there must be a
+//   unique Location with that id.
+// - There is an optional N->1 relationship from locations to
+//   mappings. For every nonzero Location.mapping_id there must be a
+//   unique Mapping with that id.
+
+syntax = "proto3";
+
+package opentelemetry.proto.profiles.v1.alternatives.pprofextended;
+
+import "opentelemetry/proto/common/v1/common.proto";
+
+option csharp_namespace = "OpenTelemetry.Proto.Profiles.V1.Alternatives.PprofExtended";
+option go_package = "go.opentelemetry.io/proto/otlp/profiles/v1/alternatives/pprofextended";
+
+// Represents a complete profile, including sample types, samples,
+// mappings to binaries, locations, functions, string table, and additional metadata.
+message Profile {
+  // A description of the samples associated with each Sample.value.
+  // For a cpu profile this might be:
+  //   [["cpu","nanoseconds"]] or [["wall","seconds"]] or [["syscall","count"]]
+  // For a heap profile, this might be:
+  //   [["allocations","count"], ["space","bytes"]],
+  // If one of the values represents the number of events represented
+  // by the sample, by convention it should be at index 0 and use
+  // sample_type.unit == "count".
+  repeated ValueType sample_type = 1;
+  // The set of samples recorded in this profile.
+  repeated Sample sample = 2;
+  // Mapping from address ranges to the image/binary/library mapped
+  // into that address range.  mapping[0] will be the main binary.
+  repeated Mapping mapping = 3;
+  // Locations referenced by samples via location_indices.
+  repeated Location location = 4;
+  // Array of locations referenced by samples.
+  repeated int64 location_indices = 15;
```

Comment:
`location_indices` stores location ids in one array, and then later samples refer to slices from that array, e.g:

```yaml
location_indices:
  - 0
  - 1
  - 2
sample:
  - locations_start_index: 0
    locations_length: 3
    value:
      - 100
  - locations_start_index: 0
    locations_length: 2
    value:
      - 200
```

if location 0 is `foo`, 1 is `bar` and 2 is `baz`, first sample refers to `foo;bar;baz` and the second one refers to `foo;bar`

Reply:
It's also covered in **Relationship Between Samples And Locations** section: https://github.com/open-telemetry/oteps/blob/c9051417675e31f83771c4881339b3b39091061e/text/profiles/0239-profiles-data-model.md#relationship-between-samples-and-locations

Reply:
Ah, sorry, the reason I got confused is because of this comment:

```
  // locations_start_index along with locations_length refers to to a slice of locations in Profile.location.
  // Supersedes location_index.
  uint64 locations_start_index = 7;
```

Based on this I had assumed that `locations_start_index` directly points to an entry in the `Profile.location` list. Maybe this comment could be adjusted to mention the indirection via `location_indices`?